core/src/test/java/org/sql2o/converters/LocalDateConverterTest.java

```diff
@@ -0,0 +1,189 @@
+        Assert.assertEquals(LocalDate.class, localDates.get(0).getClass());
+    }
+
+
+    @Test
+    public void givenValidLocalDate_whenLocalDateSavedSuccessfully_thenTableHas2RecordsOfLocalDateInstance() {
+
+        // Given
+        LocalDate localDate = LocalDate.now();
+        String sqlScriptOfInsertData = "insert into local_date_converter(created_at) values (:created_at)";
+        try (Connection connection = sql2o.open(); Query query = connection.createQuery(sqlScriptOfInsertData)) {
+            query
+                .addParameter("created_at", localDate)
+                .executeUpdate();
+        }
+
+        // When
+        List<LocalDate> localDates;
+        String sqlScriptOfFindAll = "select created_at from local_date_converter";
+        try (Connection connection = sql2o.open(); Query query = connection.createQuery(sqlScriptOfFindAll)) {
+
+            List<MockLocalDate> mockLocalDates = query.executeAndFetch(MockLocalDate.class);
+
+            localDates = mockLocalDates.stream()
+                .map(MockLocalDate::getLocalDate)
+                .collect(Collectors.toList());
+        }
+
+        // Then
+        Assert.assertNotNull(localDates);
+        Assert.assertEquals(2, localDates.size());
+        Assert.assertEquals(LocalDate.class, localDates.get(0).getClass());
+        Assert.assertEquals(LocalDate.class, localDates.get(1).getClass());
+    }
+
+
+    private static class MockLocalDate {
+
+        private LocalDate localDate;
+
+        @SuppressWarnings("It's seems unused but It's used by sql2o internally.")
+        public MockLocalDate() {
+        }
+
+        public MockLocalDate(LocalDate localDate) {
+            this.localDate = localDate;
+        }
+
+        public LocalDate getLocalDate() {
+            return localDate;
+        }
+
+    }
+
+    private static class MockLocalDateConverter implements Converter<MockLocalDate> {
+
+        @Override
+        public MockLocalDate convert(Object dateObject) throws ConverterException {
+
+            if (dateObject == null) {
+                return null;
+            }
+
+            try {
+                LocalDate localDate = LocalDate.parse(dateObject.toString().split(" ")[0]);
+                return new MockLocalDate(localDate);
+            } catch (IllegalArgumentException exception) {
+                String dateObjectClassName = dateObject.getClass().getName();
+                String localDateClassName = LocalDate.class.getName();
+                throw new ConverterException(
+                    String.format("Don't know how to convert from type '%s' to type '%s'", dateObjectClassName, localDateClassName),
+                    exception
+                );
+            }
+        }
+
+        @Override
+        public Object toDatabaseParam(MockLocalDate MockLocalDate) {
+            LocalDate localDate = MockLocalDate.getLocalDate();
+            return Date.valueOf(localDate);
+        }
+
+    }
```

Comment:
Why is this copy of the converter here? The test should test the actual converter, not a copy/mock of it. This should be removed.

Reply:
Because a LocalDateTime object cannot be instantiated with the `new` keyword like a standard Date object, I wanted to manually create my own object to use during the testing phase. Although we don't encounter this problem during application runtime, I had to apply this method specifically to run tests for this application. Without applying this method, the error falls into the case at line 24 of `ReflectionObjectConstructorFactory.java` and throws an exception. I didn't want to add a specific if case for LocalDateTime here. If you have a different solution, I would like to implement it.

Reply:
Sorry, but that doesn't make sense to me. To understand your statement above, I checked out your code, removed the `MockLocalDateConverter`, fixed the SELECT query used in the test and ran the tests. And they ran successfully! The converters you registered on lines 95 and 96 in `org.sql2o.converters.Convert.java` are doing their job. 

The `MockLocalDateConverter` and `MockLocalDateTimeConverter` must be removed.

A test, only testing test functionality, has no value.

Reply:
Can you share your code with me? How can change SELECT query?

Reply:
UUID Converters are same with my code, but it was merged. I don't understand yet

Reply:
If I only update the `ReflectionObjectConstructorFactory` class as follows, the code works.

```java
public class ReflectionObjectConstructorFactory implements ObjectConstructorFactory {
    public ObjectConstructor newConstructor(final Class<?> clazz) {
        try {

            if (clazz.equals(LocalDate.class)) {
                return LocalDate::now;
            }

            if (clazz.equals(LocalDateTime.class)) {
                return LocalDate::now;
            }

            final Constructor<?> ctor = clazz.getDeclaredConstructor();
            return () -> {
                try {
                    return ctor.newInstance((Object[]) null);
                } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
                    throw new Sql2oException("Could not create a new instance of class " + clazz, e);
                }
            };
        } catch (Exception exception) {
            if (clazz.getEnclosingClass() == null) {
                throw new Sql2oException("Could not find parameter-less constructor of class " + clazz, exception);
            } else {
                throw new Sql2oException("Could not find parameter-less constructor of class " + clazz + ", if your pojo is a nested class, you could try to make it static.", exception);
            }
        }
    }
}
```

Reply:
I am having trouble understanding why the above should be necessary. All tests run without that change. For what specific is this needed?

Reply:
Ahh, I finally think I understand why you have the above problem! 

If you run;
``` java
try (final Connection con = sql2o.open()) {
    final LocalDateTime localDateTime = con.createQuery("...SQL...").executeAndFetchFirst(LocalDateTime.class);
}
```

you will get an exception telling you that `LocalDateTime` doesn't have a parameterless constructor. That is what sql2o has the `executeScalar` method for.

This works:
``` java
try (final Connection con = sql2o.open()) {
    final LocalDateTime localDateTime = con.createQuery("...SQL...").executeScalar(LocalDateTime.class);
}
```

Reply:
Yes, this part escaped my notice, I have made the necessary arrangements ✅